### ISSUE TITLE
"Failed to run application on the device" on newly initialized React Native CLI sample app

### ISSUE THREAD

**Issue by samuelneff** [wix/Detox#2101]
**Describe the bug**
I have a newly created app from React Native CLI and added Detox with `detox init`, followed instructions to configure Android, but Detox fails to start the app on the Android emulator.

`detox -c android build` ran ok and when I run `detox -c android test` it does start the emulator, but the application itself never starts.

The application does start when run through `react-native run-android`.

It gives this error:

```
$ ./node_modules/.bin/detox -c android test
detox[37289] INFO:  [test.js] configuration="android" reportSpecs=true readOnlyEmu=false 
    useCustomLogger=true forceAdbInstall=false DETOX_START_TIMESTAMP=1591313397594 
    node_modules/.bin/jest --config e2e/config.json '--testNamePattern=^((?!:ios:).)*$' -- 
    maxWorkers 1 android test
detox[37290] INFO:  [DetoxServer.js] server listening on localhost:62332...
detox[37290] ERROR: [DetoxExportWrapper.js/DETOX_INIT_ERROR] 
 DetoxRuntimeError: Failed to run application on the device

HINT: Most likely, your tests have timed out and called detox.cleanup() while it was 
waiting for "ready" message (over WebSocket) from the instrumentation process.
```

Likely related adb logcat lines (full logcat below):

```
06-05 10:59:05.618  2036  2070 I ActivityManager: Force stopping com.reactnativetypescriptredux appid=10170 user=-1: deletePackageX
06-05 10:59:05.619  2036  2070 I ActivityManager: Killing 8148:com.reactnativetypescriptredux/u0a170 (adj 0): stop com.reactnativetypescriptredux
06-05 10:59:05.624  2036  2070 W ActivityManager: Crash of app com.reactnativetypescriptredux running instrumentation ComponentInfo{com.reactnativetypescriptredux.test/android.test.InstrumentationTestRunner}
```

Here is my config:

```json
{
  "testRunner": "jest",
  "runnerConfig": "e2e/config.json",
  "configurations": {
    "android": {
      "type": "android.emulator",
      "binaryPath": "android/app/build/outputs/apk/debug/app-debug.apk",
      "build": "cd android; ./gradlew assembleDebug assembleAndroidTest -DtestBiuldType=debug; cd -",
      "device": {
        "avdName": "Pixel_3"
      }
    }
  }
}
```

**To Reproduce**

- [X] I have tested this issue on the latest Detox release and it still reproduces

Steps to reproduce:
1.  Create new blank app with `react-native init`
2. Add detox with `detox init`
3. Update config per above
4. Build with `detox -c android build`
5. Attempt to run with `detox -c android test`

**Expected behavior**
Both the React Native app and Detox are right out of the box, so I would expect them to just run fine.

**Device and Verbose Detox Logs**

Detox logs: https://controlc.com/80359692

adb logcat: https://controlc.com/c97a8680

**Environment**
- Detox: 16.7.0
- React Native: 0.62.2
- Node: v12.17.0
- Device: Pixel 3, x86, API 29, Android 10 (also tried with API 28 / Android 9, same)
- OS: MacOS 10.15.4
- Test-runner: Jest
- Android Studio: 3.6.3

**Comment by samuelneff** [comment]
I missed some steps on the Android setup page, sorry.

**Comment by d4vidi** [comment]
@samuelneff could you please elaborate? 'anything we could improve in the guide?

**Comment by samuelneff** [comment]
Once I slowed down and read them all, the docs were fine.

The only thing I would say is to more clearly separate iOS and Android docs. The Android docs partially depend on the iOS docs but it's not clear which parts. If instead you had up front "Getting Started - iOS" and "Getting Started - Android" without either one relying on the other.

Thanks for the great product. Once I got it running, it's amazing!
